extensions/amp-auto-ads/0.1/test/test-placement.js

```diff
@@ -608,16 +608,18 @@ describes.realWin(
         });
       });
 
-      it('should set the correct attributes for responsive enabled ads using amp-ad responsive', () => {
+      it('should set the full-with responsive attributes for responsive enabled mobile users.', () => {
         const anchor = doc.createElement('div');
         anchor.id = 'anId';
         container.appendChild(anchor);
 
-        const mutator = Services.mutatorForDoc(anchor);
-        env.sandbox.stub(mutator, 'requestChangeSize').callsFake(() => {
-          return Promise.resolve();
-        });
-        env.sandbox.stub(mutator.viewport_, 'getWidth').callsFake(() => 2000);
+        const viewportMock = env.sandbox.mock(
+          Services.viewportForDoc(env.win.document)
+        );
+        viewportMock
+          .expects('getSize')
+          .returns({width: 411, height: 823})
```

Comment:
It's good practice to test the boundary conditions. So in this scenario, we should have test cases testing width=487 and another for width=488, and then maybe a couple more test cases that test extreme values too just to check it still works as intended.

There is presumably a reason you chose 488 as the threshold, so you should have tests that will fail if someone inadvertently changes that value.

Reply:
Thanks. Updated the width in the test case as you suggested.